aggregator/flush_mgr.go

```diff
@@ -210,7 +205,22 @@ func (mgr *flushManager) Close() error {
 	return nil
 }
 
-func (mgr *flushManager) bucketForWithLock(l PeriodicFlusher) (*flushBucket, error) {
+func (mgr *flushManager) findOrCreateBucketWithLock(l PeriodicFlusher) (*flushBucket, error) {
+	bucket, err := mgr.findBucketWithLock(l)
+	if err == nil || err != errBucketNotFound {
```

Comment:
nit: seems odd to me to combine these two cases (though I can see why its valid). Perhaps it would cleaner to separate them out?

```
if err == nil {
  return bucket, nil
} else if err != errBucketNotFound {
  return nil, err
}
```

Reply:
Sure sounds good.

Reply:
you can also get rid of that else

Reply:
Sure